app/controllers/autocomplete_controller.rb

```diff
@@ -0,0 +1,33 @@
+class AutocompleteController < ApplicationController
+  def github_repos
+    @repos = search_github_repos(params[:query])
+
+    respond_to do |format|
+      format.html { render partial: 'autocomplete/repository_suggestions', locals: { repos: @repos } }
+    end
+  end
+
+  private
+
+  def required_scopes
```

Comment:
the `#required_scopes` method is used to request more scopes for the user if necessary. In this case it doens't really make sense, since it's happening in the context of an ajax request.